crates/broker/src/submitter.rs

```diff
@@ -167,31 +182,56 @@ where
                     .context("Failed to get order journal from prover")?
                     .context("Order proof Journal missing")?;
 
-                // NOTE: We assume here that the order execution ended with exit code 0.
-                let order_claim =
-                    ReceiptClaim::ok(order_img_id.0, MaybePruned::Pruned(order_journal.digest()));
-                let order_claim_index = aggregation_state
-                    .claim_digests
-                    .iter()
-                    .position(|claim| *claim == order_claim.digest())
-                    .ok_or(anyhow!(
-                        "Failed to find order claim {order_claim:x?} in aggregated claims"
-                    ))?;
-                let order_path = risc0_aggregation::merkle_path(
-                    &aggregation_state.claim_digests,
-                    order_claim_index,
-                );
-                tracing::debug!(
-                    "Merkle path for order {order_id:x} : {:x?} : {order_path:x?}",
-                    order_claim.digest()
-                );
-                let set_inclusion_receipt = SetInclusionReceipt::from_path_with_verifier_params(
-                    order_claim,
-                    order_path,
-                    inclusion_params.digest(),
-                );
-                let seal =
-                    set_inclusion_receipt.abi_encode_seal().context("Failed to encode seal")?;
+                let seal = match Selector::from_bytes(order_request.requirements.selector.into())
+                    .context("Failed to parse selector")?
+                {
+                    Selector::FakeReceipt | Selector::SetVerifierV0_2 => {
```

Comment:
Here as well, it's confusing the use "FakeReceipt" to mean "any selector". Would it be possible to represent the selector field with an `Option` in Rust, for instance?